.github/workflows/report-failure.yaml

```diff
@@ -0,0 +1,87 @@
+# This workflow is intended to be called by workflows in our various software
+# repos to have shared workflow failure reporting in the form of GitHub issues.
+on:
+  workflow_call:
+    inputs:
+      failure:
+        description: Whether to create an issue indicating a failed workflow run, or keep an existing issue open.
+        type: boolean
+        required: true
+      success:
+        description: Whether to close any open issues created by this workflow.
+        type: boolean
+        required: true
+      title:
+        description: Title for issues opened by this workflow.
+        type: string
+        required: true
+    secrets:
+      token:
+        description: A GitHub access token with permission to create and modify GitHub issues.
+        required: true
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Extract workflow link
+        run: |
+          workflow_path=$(echo "${{ github.workflow_ref }}" | sed -E 's|^.*/(\.github/workflows/[^@]+)@\S*$|\1|')
+          echo "workflow_link=https://github.com/${{ github.repository }}/blob/${{ github.sha }}/$workflow_path" >> "$GITHUB_ENV"
+
+      - name: Create or close issue
+        run: |
+          # Search for issues opened by this job.
+          matching_issues=$(gh issue list \
+            --state open \
+            --author '@me' \
+            --json 'number,title' \
+            --jq 'map(select(.title == "${{ inputs.title }}") | .number)')
```

Comment:
_non-blocking question_

Should there be some filter on the branch of the workflow? Or I guess the calling workflow can control if `report-failure.yaml` only gets run on the default branch...

Reply:
I think that's outside the scope of this reusable workflow. There could be a valid reason for the calling workflow to open issues regardless of branch. Yes, I've encoded this in the calling workflow for [conda-base](https://github.com/nextstrain/conda-base/blob/a4c8ddaa6b3b6c8c9e0b9706a208f0bb015025d8/.github/workflows/installation.yaml#L47), [cli](https://github.com/nextstrain/cli/blob/ad12f213a8bcae7b94a193956ea2dff6713a4013/.github/workflows/ci.yaml#L394) PRs.

Reply:
Ah yeah that makes sense. I guess the title can include the branch of the workflow to make sure this matches issues per branch. 

Maybe the `inputs.title.description` should explicitly state that the title is used for issue matching.

Reply:
Done as part of cfa7538fb6ed08168983b8964f2299bdd8111a99